src/main/java/com/eternalcode/parcellockers/ParcelCache.java

```diff
@@ -0,0 +1,22 @@
+package com.eternalcode.parcellockers;
+
+import com.eternalcode.parcellockers.parcel.Parcel;
+import com.eternalcode.parcellockers.parcel.ParcelLocker;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ParcelCache {
```

Comment:
mam wrażenie że zawartość tej klasy może wylądować bez problemowo w ParcelManagerze,
EDIT: zobaczyłem że używasz uuid z cache jako klucza do porównywań, ta lista będzie kilka razy wolniejsza jeżeli nie zrobił byś sobie Map<UUID, ParcelCache> cache = new HashMap<>();

Reply:
No tylko że wtedy implementacja repozytoriów będzie zależna od managera, a raczej nie chcę tak robić, bo to właśnie manager ma być zależny od implementacji, bo obecnie przy wywołaniu dowolnej akcji związanych z DB, cache na bieżąco się aktualizuje